lib/membrane_opus/parser/frame_lengths.ex

```diff
@@ -0,0 +1,192 @@
+defmodule Membrane.Opus.Parser.FrameLengths do
+  @moduledoc false
+  # Helper for Membrane.Opus.Parser for determining frame lengths
+
+  # given a frame length, return opus binary encoded length
+  @spec encode_frame_length(non_neg_integer) :: binary
+  def encode_frame_length(length) do
+    if length < 252 do
+      <<length::size(8)>>
+    else
+      <<252 + rem(length - 252, 4)::size(8), div(length - 252, 4)::size(8)>>
+    end
+  end
+
+  # returns ordered list of frame lengths and header length
+  @spec parse_frame_lengths(0..3, binary, boolean) :: {[non_neg_integer], pos_integer}
+  def parse_frame_lengths(frame_packing, data_without_toc, self_delimiting?) do
+    case frame_packing do
+      # there is one frame in this packet
+      0 ->
+        code_zero_lengths(data_without_toc, self_delimiting?)
+
+      # there are two equal-length frames in this packet
+      1 ->
+        code_one_lengths(data_without_toc, self_delimiting?)
+
+      # there are two non-equal-length frames in this packet
+      2 ->
+        code_two_lengths(data_without_toc, self_delimiting?)
+
+      # there are two or more frames of arbitrary size
+      3 ->
+        code_three_lengths(data_without_toc, self_delimiting?)
+    end
+  end
+
+  @spec code_zero_lengths(binary, boolean) :: {[non_neg_integer], pos_integer}
+  defp code_zero_lengths(data_without_toc, self_delimiting?) do
+    if self_delimiting? do
```

Comment:
i don't think so? you need to know:

1) where the head ends
2) what the length of the last frame is, since the frame length can be either 1 or 2 bytes

Reply:
Ok, if it's hard to achieve let's leave it as is